### PR TITLE
Component docs

### LINKED ISSUE #34: Add more detailed component reference
Requires #30

### PR DESCRIPTION
**Pull request by forman**
Closes #34

### CODE REVIEW

File: chartlets.py/chartlets/component.py [hunk 1]
```diff
@@ -5,16 +5,30 @@
...
+    """HTML `value` property. Required for specific components."""
+
     style: dict[str, Any] | None = None
+    """HTML `style` property. Optional."""
+
     # We may add more here later
     #
     # Special non-HTML properties
     label: str | None = None
+    """Optional label used by many specific components."""
```
Comment (b-yogesh) [inline chartlets.py/chartlets/component.py:28]:
```suggestion
    """Label used by many specific components. Optional """
```

File: chartlets.py/chartlets/component.py [hunk 2]
```diff
@@ -5,16 +5,30 @@
...
+
     children: list["Component"] | None = None
+    """Optional children used by many specific components."""
```
Comment (b-yogesh) [inline chartlets.py/chartlets/component.py:31]:
```suggestion
    """Children used by many specific components. Optional """
```

File: chartlets.py/chartlets/components/box.py [hunk 3]
```diff
@@ -5,4 +5,11 @@
 
 @dataclass(frozen=True)
 class Box(Container):
-    pass
+    """The Box component is a generic container for grouping other components.
+    It's a fundamental building block. Think of it as an HTML `<div>` element.
+
+    Use the `style` attribute how to layout the box and its child components.
```
Comment (b-yogesh) [inline chartlets.py/chartlets/components/box.py:11]:
```suggestion
    Use the `style` attribute to layout the box and its child components.
```

File: chartlets.py/chartlets/components/typography.py [hunk 4]
```diff
@@ -5,4 +5,14 @@
 
 @dataclass(frozen=True)
 class Typography(Component):
+    """Use typography to present your design and content as clearly
+    and efficiently as possible."""
+
     text: str | None = None
+    """Optional text to be displayed."""
```
Comment (b-yogesh) [inline chartlets.py/chartlets/components/typography.py:12]:
```suggestion
    """Text to be displayed. Optional"""
```

File: chartlets.py/chartlets/components/typography.py [hunk 5]
```diff
@@ -5,4 +5,14 @@
...
+
+    color: str | None = None
+    """The color of the component."""
+
+    variant: str | None = None
+    """Applies the theme typography styles."""
```
Comment (b-yogesh) [inline chartlets.py/chartlets/components/typography.py:18]: Did you mean `Applies the theme to typography styles` or `Applies the theme/typography styles`?

**Review by b-yogesh — CHANGES_REQUESTED** [review]
Looks fine. Please have a look at the suggestions.

File: chartlets.py/chartlets/components/typography.py (hunk 5 above)
Comment (forman) [reply chartlets.py/chartlets/components/typography.py:18]: Copied from https://mui.com/material-ui/api/typography/

**Review by b-yogesh — APPROVED** [review]

**Pull request merged**